### PR TITLE
MANT: Remove several compiler warnings

### PR DESCRIPTION
**Pull request by jaimefrio**

### CODE REVIEW

File: numpy/random/mtrand/distributions.c [hunk 1]
```diff
@@ -315,7 +316,7 @@ long rk_binomial_btpe(rk_state *state, long n, double p)
     v = v*(u-p3)*lamr;
 
   Step50:
-    k = fabs(y - m);
+    k = labs(y - m);
```
Comment (charris) [inline numpy/random/mtrand/distributions.c]: There is the corner case where `labs(y - m)` is negative, so maybe the safer thing here is to cast `y - m` to float.

File: numpy/core/src/umath/simd.inc.src [hunk 2]
```diff
@@ -27,6 +27,19 @@
...
+{
+#if (NPY_SIZEOF_INTP <= NPY_SIZEOF_INT)
+    return abs(x);
+#elif (NPY_SIZEOF_INTP <= NPY_SIZEOF_LONG)
+    return labs(x);
+#else
+    return llabs(x);
+#endif
+}
+
```
Comment (charris) [inline numpy/core/src/umath/simd.inc.src]: Hmm, certainly no worse than it was ;) I'm thinking about the corner case of `abs` not working for the most negative value of signed integers, but that is a whole other can of worms.

File: numpy/random/mtrand/distributions.c (hunk 1 above)
Comment (charris) [inline numpy/random/mtrand/distributions.c]: Hmm, k is `long`, also gets squared further down (overflow?). I quess this is no worse than it was, but the whole routine might be less than perfect.

**Comment by charris** [comment]
The whole `abs` thing for integers could probably use an audit, but this seems no worse than it was. Thanks Jaime.

**Comment by cgohlke** [comment]
`llabs` is not available in Visual Studio 2008.

**Comment by charris** [comment]
Oops.

**Pull request merged**